aries_cloudagent/protocols/out_of_band/v1_0/manager.py

```diff
@@ -279,6 +284,99 @@ async def create_invitation(
                     routing_keys=[],
                 ).serialize()
 
+        elif did_peer_4 or did_peer_2:
```

Comment:
This function is getting quite long and is difficult to track logic. I would suggest at some point we refactor this out into smaller and more digestible blocks.

Reply:
Yes I agree, I think that's a task for the next time a DID method needs to be supported for connection reuse

I chatted with Stephen about this general topic and we agreed to put it off.  The next DID(s) - that we know of - that need to support connection reuse are all public, so we should be good for now ...